### PR TITLE
Allow Payjoin for wallet receive addresses

### PR DESCRIPTION
**Pull request by Kukks**
This PR 
* enables payjoin support for wallet receive addresses not tied to invoices. 
* fixes a bug in Wallet Send where if a bip21 did not have an amount it would fail
* adds bip21 to wallet address greenfield api

### CODE REVIEW

File: BTCPayServer/Controllers/GreenField/StoreOnChainWalletsController.cs [hunk 1]
```diff
@@ -22,6 +23,7 @@
 using NBXplorer;
 using NBXplorer.Models;
 using Newtonsoft.Json.Linq;
+using Org.BouncyCastle.Ocsp;
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/GreenField/StoreOnChainWalletsController.cs:26]: remove this

File: BTCPayServer/Controllers/GreenField/StoreOnChainWalletsController.cs [hunk 2]
```diff
@@ -122,9 +124,18 @@ public async Task<IActionResult> GetOnChainWalletReceiveAddress(string storeId,
             {
                 return BadRequest();
             }
+
+            var bip21 = network.GenerateBIP21(kpi.Address.ToString(), null);
+            var allowedPayjoin = derivationScheme.IsHotWallet && Store.GetStoreBlob().PayJoinEnabled;
+            if (allowedPayjoin)
+            {
+               bip21 +=
+                   $"?{PayjoinClient.BIP21EndpointKey}={Request.GetAbsoluteUri(Url.Action("Submit", "PayJoinEndpoint", new {cryptoCode}))}";
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/GreenField/StoreOnChainWalletsController.cs:133]: Can you use `nameof()` for Submit?

**Comment by NicolasDorier** [comment]
Nice, code review ACK, though I think you should add in the API a way to disable payjoin for the newly generated address.

File: BTCPayServer/Controllers/GreenField/StoreOnChainWalletsController.cs (hunk 2 above)
Comment (Kukks) [reply BTCPayServer/Controllers/GreenField/StoreOnChainWalletsController.cs:133]: done

File: BTCPayServer/Controllers/GreenField/StoreOnChainWalletsController.cs (hunk 1 above)
Comment (Kukks) [reply BTCPayServer/Controllers/GreenField/StoreOnChainWalletsController.cs:26]: done

**Comment by MaiHistoryCompanyLimited** [comment]
{#2425}

**Comment by Kukks** [comment]
> {#2425}

Stop spamming all out PRs.

**Comment by Kukks** [comment]
> Nice, code review ACK, though I think you should add in the API a way to disable payjoin for the newly generated address. (and default to payjoin enabled)

We can do this if someone requests it IMO. It currently checks if the user has payjoin enabled in their wallet

**Pull request merged**